test/Collector.test.ts

```diff
@@ -0,0 +1,734 @@
+    defaultOwnerAddr: string,
+    defaultTokens: string[],
+    defaultPartners: Partner[],
+    defaultRemainderDestinationAddr: string
+  ): CollectorDeployFunction =>
+  ({
+    ownerAddr = defaultOwnerAddr,
+    tokens = defaultTokens,
+    partners = defaultPartners,
+    remainderDestinationAddr = defaultRemainderDestinationAddr,
+  }: CollectorDeployParams) =>
+    collectorFactory.deploy(
+      ownerAddr,
+      tokens,
+      partners,
+      remainderDestinationAddr
+    );
+
+function checkPartnerBalanceForToken(
+  partners: Partner[],
+  tokenToWithdraw: FakeContract<ERC20>
+) {
+  for (let partnerIndex = 0; partnerIndex < partners.length; partnerIndex++) {
+    expect(
+      tokenToWithdraw.transfer,
+      `Partner[${partnerIndex}] balance for token[${tokenToWithdraw.address}]`
+    ).to.have.been.calledWith(
+      partners[partnerIndex].beneficiary,
+      PARTNER_SHARES[partnerIndex]
+    );
+  }
+}
+
+describe('Collector', function () {
+  let owner: SignerWithAddress;
+  let collectorFactory: MockContractFactory<Collector__factory>;
+  let fakeERC20Tokens: FakeContract<ERC20>[];
+  let partners: Partner[];
+  let remainderDestination: SignerWithAddress;
+  let partner1: SignerWithAddress;
+  let partner2: SignerWithAddress;
+  let partner3: SignerWithAddress;
+  let partner4: SignerWithAddress;
+
+  beforeEach(async function () {
+    [owner, partner1, partner2, partner3, partner4, remainderDestination] =
+      await ethers.getSigners();
+    partners = await buildPartners(
+      [partner1, partner2, partner3, partner4],
+      PARTNER_SHARES
+    );
+
+    fakeERC20Tokens = await Promise.all(
+      Array.from(Array(2)).map(async (fakeToken: FakeContract<ERC20>) => {
+        fakeToken = await smock.fake<ERC20>('ERC20');
+        fakeToken.transfer.returns(true);
+
+        return fakeToken;
+      })
+    );
+
+    collectorFactory = await smock.mock<Collector__factory>('Collector');
+  });
+
+  describe('constructor()', function () {
+    it('Should deploy a Collector with single token', async function () {
+      partners = await buildPartners(
+        [partner1, partner2, partner3, partner4],
+        PARTNER_SHARES
+      );
+      const expectedTokens = fakeERC20Tokens.map(({ address }) => address)[0];
+      const expectedOwnerAddress = owner.address;
+
+      const collector: MockContract<Collector> = await collectorFactory.deploy(
+        expectedOwnerAddress,
+        [expectedTokens],
+        partners,
+        remainderDestination.address
+      );
+
+      const actualOwnerAddress = await collector.owner();
+      const actualTokens = await collector.getTokens();
+
+      expect(actualOwnerAddress, 'Failed to set the owner').to.equal(
+        expectedOwnerAddress
+      );
+      expect(actualTokens.toString(), 'Failed to set tokens').to.equal(
+        expectedTokens.toString()
+      );
+    });
+
+    it('Should deploy a Collector with multiple tokens', async function () {
```

Comment:
I think that we could include the unit testing if we are going to do a deployment without tokens.

Reply:
Thank you for highlighting it, done